plugins/inputs/jolokia2/README.md

```diff
@@ -0,0 +1,113 @@
+
+  # Supply a 'path' to collect a simple scalar value called 'Uptime'.
+  [[inputs.jolokia2.metric]]
+    name  = "jvm_runtime"
+    mbean = "java.lang:type=Runtime"
+    paths = ["Uptime"]
+
+  # More complex values may be collected as well.
+  [[inputs.jolokia2.metric]]
+    name  = "jvm_memory"
+    mbean = "java.lang:type=Memory"
+    paths = ["HeapMemoryUsage", "NonHeapMemoryUsage", "ObjectPendingFinalizationCount"]
+
+  # Use mbean object patterns to create distinct series.
+  # TODO: needs simple example
+
+  # By default, all mbean keys are added as tags
+  # Use 'tag_keys' to specify exactly the keys to add as tags.
+  [[inputs.jolokia2.metric]]
+    name     = "jvm_garbage_collector"
+    mbean    = "java.lang:name=*,type=GarbageCollector"
+    paths    = ["CollectionTime", "CollectionCount"]
+    tag_keys = ["name"]
+
+  # Use 'untag_keys' to specify just the tags to remove.
+  [[inputs.jolokia2.metric]]
+    name       = "jvm_memory_pool"
+    mbean      = "java.lang:name=*,type=MemoryPool"
+    paths      = ["Usage", "PeakUsage, "CollectionUsage"]
+    untag_keys = ["type"]
+
+  # Use simple substitutions to alter field prefixes with mbean properties.
+  [[inputs.jolokia2.metric]]
+    name         = "kafka_topic"
+    mbean        = "kafka.server:name=*,topic=*,type=BrokerTopicMetrics"
+    field_prefix = "$1"
+    tag_keys   = ["topic"]
+
+  # In some cases, it makes sense to substitute field names for mbean properties.
+  [[inputs.jolokia2.metric]]
+    name       = "kafka_log"
+    mbean      = "kafka.log:name=*,partition=*,topic=*,type=Log"
+    field_name = "$1"
+    tag_keys = ["topic", "partition"]
+```
+
+To specify timeouts for slower/over-loaded clients:
+
+```
+[[inputs.jolokia2]]
+  [inputs.jolokia2.agents]
+    urls = ["http://kafka:8080/jolokia"]
+
+    # The amount of time to wait for any requests made by this client.
+    # Includes connection time, any redirects, and reading the response body.
+    # (default is 5s)
+    response_timeout = "10s"
+```
+
+To specify SSL options, add details to the `agents` configuration:
+
+```
+[[inputs.jolokia2]]
+  [inputs.jolokia2.agents]
+    urls = [
+      "https://kafka:8080/jolokia",
+    ]
+    #username = ""
+    #password = ""
+    ssl_ca   = "/var/private/ca.pem"
+    ssl_cert = "/var/private/client.pem"
+    ssl_key  = "/var/private/client-key.pem"
+    #insecure_skip_verify = false
+```
+
+To interact with agents via a Jolokia proxy, use a `proxy` configuration instead:
+
+```
+[[inputs.jolokia2]]
+  [inputs.jolokia2.proxy]
+    url = "https://proxy:8080/jolokia"
+    response_timeout = "10s"
+    #default_target_username = ""
+    #default_target_password = ""
+    ssl_ca   = "/var/private/ca.pem"
+    ssl_cert = "/var/private/client.pem"
+    ssl_key  = "/var/private/client-key.pem"
+
+    [[inputs.jolokia2.proxy.target]]
+      url = "service:jmx:rmi:///jndi/rmi://targethost:9999/jmxrmi"
+      #username = ""
+      #password = ""
+```
```

Comment:
+1, curious if the problem of collapsing metrics with long paths is fixed.
ATM we're using Cassandra jolokia plugin and it seems that plugin collapses `LastGcInfo` for all heap spaces into one. For instance:
```
metrics = [
"/java.lang:type=GarbageCollector,name=G1%20Young%20Generation/LastGcInfo/memoryUsageAfterGc/G1%20Eden%20Space",
"/java.lang:type=GarbageCollector,name=G1%20Young%20Generation/LastGcInfo/memoryUsageAfterGc/G1%20Survivor%20Space",
"/java.lang:type=GarbageCollector,name=G1%20Young%20Generation/LastGcInfo/memoryUsageAfterGc/G1%20Old%20Gen",
"/java.lang:type=GarbageCollector,name=G1%20Young%20Generation/LastGcInfo/memoryUsageAfterGc/Metaspace",
"/java.lang:type=GarbageCollector,name=G1%20Young%20Generation/LastGcInfo/memoryUsageAfterGc/Compressed%20Class%20Space",
"/java.lang:type=GarbageCollector,name=G1%20Young%20Generation/LastGcInfo/memoryUsageAfterGc/Code%20Cache"
]
```
result in only following 5:
```
"javaGarbageCollector,mname=G1\\ Young\\ Generation,cassandra_host=test LastGcInfo_used=0,LastGcInfo_init=4513071104,LastGcInfo_committed=4160749568,LastGcInfo_max=-1 1487259680000000000\n"
"javaGarbageCollector,mname=G1\\ Young\\ Generation,cassandra_host=test LastGcInfo_max=-1,LastGcInfo_used=352321536,LastGcInfo_init=0,LastGcInfo_committed=352321536 1487259680000000000\n"
"javaGarbageCollector,mname=G1\\ Young\\ Generation,cassandra_host=test LastGcInfo_init=12666798080,LastGcInfo_committed=12666798080,LastGcInfo_max=17179869184,LastGcInfo_used=4759843152 1487259680000000000\n"
"javaGarbageCollector,mname=G1\\ Young\\ Generation,cassandra_host=test LastGcInfo_used=39882464,LastGcInfo_init=0,LastGcInfo_committed=40714240,LastGcInfo_max=-1 1487259680000000000\n"
"javaGarbageCollector,mname=G1\\ Young\\ Generation,cassandra_host=test LastGcInfo_init=0,LastGcInfo_committed=4587520,LastGcInfo_max=1073741824,LastGcInfo_used=4326520 1487259680000000000\n"
"javaGarbageCollector,mname=G1\\ Young\\ Generation,cassandra_host=test LastGcInfo_init=2555904,LastGcInfo_committed=44236800,LastGcInfo_max=251658240,LastGcInfo_used=41900864 1487259680000000000\n"
```

whereas they should (perhaps) have path in their names providing metric names sort of like (or in any way where path
```
"javaGarbageCollector,mname=G1\\ Young\\ Generation,cassandra_host=test memoryUsageAfterGc_Metaspace_LastGcInfo_init=2555904,memoryUsageAfterGc_Metaspace_LastGcInfo_committed=44236800,memoryUsageAfterGc_Metaspace_LastGcInfo_max=251658240,memoryUsageAfterGc_Metaspace_LastGcInfo_used=41900864 1487259680000000000\n"
```